app/assets/javascripts/3dbio_viewer/src/domain/repositories/DbModelRepository.ts

```diff
@@ -1,9 +1,10 @@
 import { DbModelCollection, DbModelType } from "../entities/DbModel";
 import { FutureData } from "../entities/FutureData";
+import { UploadData } from "../entities/UploadData";
 
 export interface DbModelRepository {
     search(options: SearchOptions): FutureData<DbModelCollection>;
-    upload(options: UploadOptions): FutureData<unknown>;
+    upload(options: UploadOptions): FutureData<UploadData>;
```

Comment:
`UploadData` is too generic for a name. It's hard to come up with good names. Maybe `AtomicStructure` for the model.